### ISSUE TITLE
Supporting `count` as a metric

### ISSUE THREAD

**Issue by adrinjalali** [fairlearn/fairlearn#737]
It's desired to know the number of data points in each group when working with `MetricFrame`, and that can be achieved by passing a function which returns the length of its input. For instance:

``` python
# report the above scores in a table.
from fairlearn.metrics import MetricFrame
import sklearn.metrics as skm
y_pred = np.asarray(df.y_pred).astype(float)
y_true = np.asarray(df.FraudLabel == "true").astype(int)
multi_metric = MetricFrame({
    'average precision':skm.average_precision_score,
    'roc-auc': skm.roc_auc_score,
    'count': lambda x, y: len(x)},
    y_true, y_pred,
    sensitive_features=df.gender)
multi_metric.by_group
```

We don't necessarily need to change anything in the code or API for this, but we should document it in the user guide and/or the API reference to let people know how it can be done.

**Comment by romanlutz** [comment]
I like this a lot. The `lambda x, y: len(x)` isn't super straightforward if you're not familiar with how to write a metric function. Should we just define the `count` function like we did with `selection_rate` so that people can import it and pass it without having to write the code themselves?

Totally agree on documenting this. I guess the tasks here would be
- [ ] to look at the user guide of `MetricFrame` and add that new `count` metric wherever we have 2 or more metrics passed to the `MetricFrame`, and write a short sentence or two on what this does (i.e., "show the number of people in the corresponding group")
- [ ] to look through notebooks and add `count` whenever we have a `MetricFrame`.

**Comment by hildeweerts** [comment]
I like the convenience of a `count` function, it makes it more beginner friendly I think. Do we need to open a separate issue for that?

**Comment by adrinjalali** [comment]
I think the same issue/pr could do both the doc and the function.

**Comment by MiroDudik** [comment]
To clarify the `count` function proposal. I'm assuming we're just talking about adding `fairlearn.metrics.count`, but not doing anything with the `MetricFrame` API right?

**Comment by adrinjalali** [comment]
> To clarify the `count` function proposal. I'm assuming we're just talking about adding `fairlearn.metrics.count`, but not doing anything with the `MetricFrame` API right?

Correct :)

**Comment by vamsidesu5** [comment]
Hey I can take this issue up!

**Comment by adrinjalali** [comment]
Sure, it's yours @vamsidesu5

**Comment by stephenrobic** [comment]
I will begin working on this issue now. Thanks for your guidance @romanlutz

**Comment by romanlutz** [comment]
Thanks @stephenrobic ! @vamsidesu5 let me know if you want to do any other issues! I'd be happy to find you something.

**Comment by vamsidesu5** [comment]
@romanlutz Hi - I was very busy the past 2 weeks and wasn't able to contribute. So sorry! Would you be able to find me another issue that would be interesting to work on?

**Comment by romanlutz** [comment]
Closing since @stephenrobic 's PR #812  was merged a week ago.
